### PR TITLE
Add release process markdown file

### PR DESCRIPTION
**Pull request by glennsong09**
Add a release process markdown file to more clearly document our release process.

### CODE REVIEW

File: release_docs/RELEASE_PROCESS.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+8. Finalize Release Notes (Release Manager)
+
+9. Package and Distribute Release (Release Manager)
+
+10. Conduct Release Retrospective (Release Manager)
+
+| Task Name     | Task Items    | Task Details | 
+| ------------- | ------------- | ---------- |
+| 3. Prepare Release Notes (Release Manager) | | |
+| | 1. Audit Release Changes (Release Manager) | Confirm that all non-trivial changes made to the source are reflected in the release notes. Verify the following: <br> [HDF5 Milestones Projects](https://github.com/HDFGroup/hdf5/milestones) <br> Each entry in [RELEASE.txt](https://github.com/HDFGroup/hdf5/blob/develop/release_docs/RELEASE.txt) traces to one or more resolved GH issues marked with FixVersion="X.Y.Z". <br> Each resolved GH milestone issue traces to an entry in [RELEASE.txt](https://github.com/HDFGroup/hdf5/blob/develop/release_docs/RELEASE.txt).<br>Each resolved GH milestone issue traces to one or more revisions to the HDF5 source. <br> Each resolved GH milestone issue traces to one or more pull requests. |
```
Comment (byrnHDF) [inline release_docs/RELEASE_PROCESS.md]: Can we put these numbered substeps under the Task before the table and then just use the Task Details as a list?

Comment (glennsong09) [reply release_docs/RELEASE_PROCESS.md]: Sorry, can I get a bit more clarification on what you want?

Comment (byrnHDF) [reply release_docs/RELEASE_PROCESS.md]: Decompose the table moving each row of a task and steps under the task # above the table
line 36 is replaced by :
 1. Audit Release Changes (Release Manager)
     - Confirm that all non-trivial changes made to the source are reflected in the release notes.
     - Verify the following:
           - [HDF5 Milestones Projects](https://github.com/HDFGroup/hdf5/milestones)
           - Each entry in [RELEASE.txt](https://github.com/HDFGroup/hdf5/blob/develop/release_docs/RELEASE.txt) traces to one or more resolved GH issues marked with FixVersion="X.Y.Z".
           -  ....
 3. Confirm and Update Known Issues (Release Manager or Software Maintainers)

File: release_docs/RELEASE_PROCESS.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+| | 2. Enable Automated Regression Testing (Test Automation Team) | Schedule and enable daily automated regression testing of the release preparation branch (i.e. hdf5_X_Y_Z). Give the 'OK' to proceed once all required tests have verified that HDF5 is functioning as intended on all target operating environments. |
+| | 3. Update .so numbers according to the report (Release Manager) | See [Updating version info (Libtool)](https://www.gnu.org/software/libtool/manual/html_node/Updating-version-info.html#Updating-version-info)  for rules to help update library version numbers. |
+| | 4. Execute release workflow | Select release build from workflow.<br>Choose the release branch<br>Change ‘Release version tag’ name to 'release' |
+| | 5. Notify HDF-Forum (Release Manager) | Send a message to the hdf-forum indicating that a pre-release source package is available for testing at **<https://gamma.hdfgroup.org/ftp/pub/outgoing/hdf5/{hdf5-X.Y.Z-pre<n>>}(look at this)** and that feedback from the user community on their test results is being accepted.|
```
Comment (lrknox) [inline release_docs/RELEASE_PROCESS.md]:
```suggestion
| | 5. Notify HDF-Forum (Release Manager) | Send a message to the hdf-forum indicating that a pre-release source package is available for testing at **<e.g. gamma.hdfgroup.org/ftp/pub/outgoing/hdf5/{hdf5-X.Y.Z-pre<n>}>(look at this)** and that feedback from the user community on their test results is being accepted.|
```

File: release_docs/RELEASE_PROCESS.md [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+Maintenance releases are always backward compatible with regards to the HDF5 file format, meaning that:
+- New libraries and command line utilities can access HDF5 files created by the previous versions of the libraries.
+
+Maintenance releases are always forward compatible with regards to the HDF5 file format, meaning that:
+- HDF5 libraries and command line utilities can access files created by future maintenance versions of the library.
+Note that maintenance releases are NOT guaranteed to be interface-compatible, meaning that, on occasion, application source code will need updated and re-compiled against a new maintenance release when the interface changes. Interface changes are only made when absolutely necessary as deemed by the HDF5 product manager(s), and interface compatibility reports are published with each release to inform customers and users of any incompatibilities in the interface.
+
+For more information on the HDF5 versioning and backward and forward compatibility issues, see the [HDF5 FAQ](http://www.hdfgroup.org/HDF5-FAQ.html) on the public website.
```
Comment (bmribler) [inline release_docs/RELEASE_PROCESS.md]:
```suggestion
For more information on the HDF5 versioning and backward and forward compatibility issues, see the [API Compatibility Macros](https://docs.hdfgroup.org/hdf5/v1_14/api-compat-macros.html) on the public website.
```

Comment (lrknox) [reply release_docs/RELEASE_PROCESS.md]: These two suggestions, when committed, should allow the link checks to pass.

File: release_docs/RELEASE_PROCESS.md [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+- Software Tester — An individual responsible for identifying, implementing, and conducting tests of software code, modules, packages, executables, and/or release binaries.
+- Test Automation Team — A team at The HDF Group responsible for overseeing the the daily automated regression tests of software maintained by the company's Sustaining Engineering program.
+
+## Tasks:
+1. Plan Maintenance Release (Product Manager | Release Manager)
+
+2. Perform Software Maintenance (Software Developers | Product Manager | Release Manager)
+
+3. Prepare Release Notes (Release Manager)
+    - Audit the sufficiency of and update the release notes.
```
Comment (hyoklee) [inline release_docs/RELEASE_PROCESS.md:36]: sufficiency of ____?

**Review by hyoklee — APPROVED** [review]

File: release_docs/RELEASE_PROCESS.md [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+| | 13. Review Test Sufficiency (Release Manager)|
+| 8. Finalize Release Notes (Release Manager) | |
+| | 1. Update Release Notes in **release** branch (Release Manager)| 
+| | 2. Update Release Notes in **support** branch (Release Manager) | 
+| 9. Package and Distribute Release | |
+| | 1. Execute release workflow | 
+| | 2. Review | 
+| | 3. Change status from Pre-Release to Release | 
+| 10. Conduct Release Retrospective (Release Manager) | |
+| | 1. Schedule Time and Solicit Comments  (Release Manager) |
```
Comment (byrnHDF) [inline release_docs/RELEASE_PROCESS.md]: Add these two line to number 10 above and then you should be able to delete this whole table.

File: release_docs/RELEASE_PROCESS.md [hunk 6]
```diff
@@ -0,0 +1,204 @@
...
+| | 6. Notify Clients (Project Leads) | 
+| | 7. Sign Binaries (Test Automation Team) | 
+| | 8. Communicate Test Assignments (Release Manager) | 
+| | 9. Perform Manual Source Testing (Software Testers) | 
+| May use binaries from GH | 10. Perform Binary Testing (Software Testers) | 
+| | 11. Report Status (Software Testers)| 
+| | 12. Resolve Issues (Release Manager) |
+| | 13. Review Test Sufficiency (Release Manager)|
+| 8. Finalize Release Notes (Release Manager) | |
+| | 1. Update Release Notes in **release** branch (Release Manager)|
```
Comment (byrnHDF) [inline release_docs/RELEASE_PROCESS.md]: I didn't see this line under Task 8 above, just support branch

File: release_docs/RELEASE_PROCESS.md [hunk 7]
```diff
@@ -0,0 +1,204 @@
...
+        - Change `default=debug` to `default=production` at the bottom of the `AS_HELP_STRING` for `--enable-build-mode`.
+        - Under `if test "X-$BUILD_MODE" = X- ; then` change `BUILD_MODE=debug` to `BUILD_MODE=production`. 
+        - Run `“sh ./autogen.sh”` to regenerate the UNIX build system files and commit the changes. (use `git status --ignored` to see the changes and `git add -f` to add all files. First delete any new files not to be committed, notably `src/H5public.h~` and `autom4te.cache/`.)
+        - `$ git push with commit message listing change steps for creating release branch`
+    - E-mail hdf5lib@hdfgroup.org to indicate that the code freeze on the release support branch (i.e. hdf5_X_Y) has been lifted and development on the next maintenance release can resume. The code freeze will remain in place on the release preparation branch (i.e. hdf5_X_Y_Z) indefinitely. 
+
+7. Perform Release Testing (Test Automation Team | Release Manager | Project Leads)
+    - Verify that source and binary distributions of HDF5 are acceptable on all target operating environments.
```
Comment (byrnHDF) [inline release_docs/RELEASE_PROCESS.md]: Should the sub-tasks be an ordered list? Or is this unnumbered list good?

Comment (glennsong09) [reply release_docs/RELEASE_PROCESS.md]: I have updated this.

File: release_docs/RELEASE_PROCESS.md (hunk 5 above)
Comment (glennsong09) [reply release_docs/RELEASE_PROCESS.md]: I have deleted the table.

File: release_docs/RELEASE_PROCESS.md (hunk 6 above)
Comment (glennsong09) [reply release_docs/RELEASE_PROCESS.md]: I moved this up as well.

File: release_docs/RELEASE_PROCESS.md [hunk 8]
```diff
@@ -0,0 +1,154 @@
...
+    1. Verify that source and binary distributions of HDF5 are acceptable on all target operating environments.
+    2. Create a page on Confluence as a sub-page of the current release version's project collaboration page (see HDF5 Maintenance Releases) to document release testing results. 
+    3. Document the test procedure that will be used for this release on the new sub-page. 
+    4. Schedule daily stand-up meetings or other regular checkpoints to assure progress on testing tasks, and inform testing team of your expectations.
+    5. Schedule and enable daily automated regression testing of the release preparation branch (i.e. hdf5_X_Y_Z). Give the 'OK' to proceed once all required tests have verified that HDF5 is functioning as intended on all target operating environments. 
+    6. See [Updating version info (Libtool)](https://www.gnu.org/software/libtool/manual/html_node/Updating-version-info.html#Updating-version-info)  for rules to help update library version numbers. 
+    7. Select release build from workflow.
+    8. Choose the release branch
+    9. Change ‘Release version tag’ name to 'release'
```
Comment (byrnHDF) [inline release_docs/RELEASE_PROCESS.md]: hdf5_X.Y.Z.P instead of release

Comment (glennsong09) [reply release_docs/RELEASE_PROCESS.md]: This has been updated.

File: release_docs/RELEASE_PROCESS.md [hunk 9]
```diff
@@ -0,0 +1,159 @@
+# Release Process
+
+## Objective:
+Release a new version of HDF software, which corrects defects, adds functionality, improves maintainability, and/or ports it to new operational environments.
+
+## Description (General):
+At regularly scheduled intervals throughout each year, The HDF Group releases a new version of each of its maintained / core software products and corresponding documentation set. This is referred to as a maintenance release. The main purpose of a maintenance release is to address defects reported against previous releases of the software, to port the software to new operating environments, and to improve maintainability by way of refactoring code and improving documentation. A maintenance release may also include new functionality in any of the product's components.
```
Comment (bmribler) [inline release_docs/RELEASE_PROCESS.md]: maintained/core vs. maintained / core?

File: release_docs/RELEASE_PROCESS.md [hunk 10]
```diff
@@ -0,0 +1,159 @@
...
+
+For more information on the HDF5 versioning and backward and forward compatibility issues, see the [API Compatibility Macros](https://docs.hdfgroup.org/hdf5/v1_14/api-compat-macros.html) on the public website.
+
+## Participants:
+- Product Manager — The individual responsible for the overall direction and development of a software product at The HDF Group.
+- Release Manager — The individual responsible for coordinating and overseeing the activities specific to releasing a software product at The HDF Group.
+- Software Developer — An individual or group responsible for performing maintenance on supported HDF software.
+- Software Tester — An individual responsible for identifying, implementing, and conducting tests of software code, modules, packages, executables, and/or release binaries.
+- Test Automation Team — A team at The HDF Group responsible for overseeing the the daily automated regression tests of software maintained by the company's Sustaining Engineering program.
```
Comment (bmribler) [inline release_docs/RELEASE_PROCESS.md]: extra "the"

File: release_docs/RELEASE_PROCESS.md [hunk 11]
```diff
@@ -0,0 +1,159 @@
...
+        - The compatibility reports are produced by the CI and are viewable in the Github [Releases/snapshot](https://github.com/HDFGroup/hdf5/releases/tag/snapshot) section.
+    6. Confirm the necessity of and approve of any interface-breaking changes. If any changes need to be reverted, task the developer who made the change to do so as soon as possible. If a change is reverted, return to the previous step and regenerate the compatibility report after the changes is made. Otherwise, continue to the next step.
+    7. Update the .so version numbers in the [config/lt_vers.am](https://github.com/HDFGroup/hdf5/blob/develop/config/lt_vers.am) file in the support branch according to [libtool's library interface version](https://www.gnu.org/software/libtool/manual/libtool.html#Versioning) scheme. 
+        - See [Updating version info (Libtool)](https://www.gnu.org/software/libtool/manual/html_node/Updating-version-info.html#Updating-version-info) for rules to help update library version numbers. 
+    8. After the release branch has been created, run `./autogen.sh` to regenerate build system files on the release branch and commit changes.
```
Comment (bmribler) [inline release_docs/RELEASE_PROCESS.md]: Suggest: "...and commit the changes."

File: release_docs/RELEASE_PROCESS.md [hunk 12]
```diff
@@ -0,0 +1,159 @@
...
+
+6. Prepare Release Branch (Release Manager)
+    1. Get the release branch ready for pre-release testing and packaging.
+    2. For all release preparation operations, the release coordinator will clone and push directly to canonical HDF5:
+        - `$ git clone ​https://github.com/HDFGroup/hdf5.git`
+    3. Create a release preparation branch (hdf5_X_Y_Z) by branching off of the support branch (hdf5_X_Y):
+        - `$ git checkout hdf5_X_Y`
+        - `$ git checkout -b hdf5_X_Y_Z`
+        - or create the new branch in GitHub GUI.
+    4. Check that required CMake files to point to the specific versions of the third-party software (szip and zlib) that they depend on.
```
Comment (bmribler) [inline release_docs/RELEASE_PROCESS.md]: Suggest: "Check that required CMake files point to..." (leave out a "to")

File: release_docs/RELEASE_PROCESS.md [hunk 13]
```diff
@@ -0,0 +1,159 @@
...
+    2. Create a page on Confluence as a sub-page of the current release version's project collaboration page (see HDF5 Maintenance Releases) to document release testing results. 
+    3. Document the test procedure that will be used for this release on the new sub-page. 
+    4. Schedule daily stand-up meetings or other regular checkpoints to assure progress on testing tasks, and inform testing team of your expectations.
+    5. Schedule and enable daily automated regression testing of the release preparation branch (i.e. hdf5_X_Y_Z). Give the 'OK' to proceed once all required tests have verified that HDF5 is functioning as intended on all target operating environments. 
+    6. Select release build from workflow.
+    7. Choose the release branch
+    8. Change ‘Release version tag’ name to 'hdf5_X.Y.Z.P'
+        - P is some pre-release number.
+    9. Send a message to the HDF forum indicating that a pre-release source package is available for testing at <e.g. gamma.hdfgroup.org/ftp/pub/outgoing/hdf5/{hdf5-X.Y.Z-pre<n>}> and that feedback from the user community on their test results is being accepted.
```
Comment (bmribler) [inline release_docs/RELEASE_PROCESS.md]: 'P' vs "pre\<n\>"?

File: release_docs/RELEASE_PROCESS.md [hunk 14]
```diff
@@ -0,0 +1,159 @@
...
+    10. Contact paying clients who are interested in testing the pre-release source package and inform them that it is available for testing and that feedback on their test results of the pre-release is appreciated.
+    11. This should be automated and currently github binaries are not signed.
+        - Follow the [How to sign binaries with digital certificates(this is missing)]() work instructions to sign each windows and mac binary package with a digital certificate.
```
Comment (bmribler) [inline release_docs/RELEASE_PROCESS.md]: Windows and Mac? (cap?)

File: release_docs/RELEASE_PROCESS.md [hunk 15]
```diff
@@ -0,0 +1,159 @@
...
+    15. Scripted Testing: 
+        - UNIX: [Scripted Binary Testing of HDF5 on UNIX systems (this is missing)]() 
+        - Windows: [Testing HDF5 Binaries(this is missing)]() 
+    16. Manual Testing (i.e. verifying correct test outcomes via visual inspection): 
+        - Use this if UNIX test script is not reporting correct results, yet binaries look OK. 
+        - UNIX: [Manual Binary Testing of HDF5 on Unix systems (this is missing)]() 
+    17. Update the test results Confluence page with status/outcome of all test assignments.
+    18. If any test source (hdf-forum, clients, internal testers, automated regression suite) identifies any issues: 
+        - a) Enter the issue in JIRA summarizing the failure if it is not already there. 
+        - b) Decide whether or not to resolve the issue for this release, or to simply document it as a known issue for this release.
```
Comment (bmribler) [inline release_docs/RELEASE_PROCESS.md]: Suggest: "Decide whether to resolve the issue for this release or to simply document..."

File: release_docs/RELEASE_PROCESS.md [hunk 16]
```diff
@@ -0,0 +1,159 @@
...
+        - Choose the release branch
+        - Change ‘Release version tag’ name to 'hdf5_X.Y.Z' 
+        - Press "Run Workflow"
+    3. Review the release files in Github
+    4. Edit the Github Release and change status to Release
+        - Change status from Pre-release to Release
+
+10. Conduct Release Retrospective (Release Manager)
+    1. Schedule time and solicit comments from retrospective
+    2. Identify issues and document them
```
Comment (bmribler) [inline release_docs/RELEASE_PROCESS.md]: Maybe a different numbering should be used for the sub-lists, for example:
1.
    a.
    b.
or
1.
    1.1
    1.2

Comment (bmribler) [reply release_docs/RELEASE_PROCESS.md]: Ah, @byrnHDF already suggested about the numbering.

Comment (byrnHDF) [reply release_docs/RELEASE_PROCESS.md]: Maybe make the tasks bold? Might help in distinguishing tasks from its steps?

File: release_docs/RELEASE_PROCESS.md (hunk 9 above)
Comment (glennsong09) [reply release_docs/RELEASE_PROCESS.md]: I have fixed this.

File: release_docs/RELEASE_PROCESS.md (hunk 10 above)
Comment (glennsong09) [reply release_docs/RELEASE_PROCESS.md]: Fixed.

File: release_docs/RELEASE_PROCESS.md (hunk 11 above)
Comment (glennsong09) [reply release_docs/RELEASE_PROCESS.md]: Updated.

File: release_docs/RELEASE_PROCESS.md (hunk 12 above)
Comment (glennsong09) [reply release_docs/RELEASE_PROCESS.md]: Fixed.

File: release_docs/RELEASE_PROCESS.md (hunk 13 above)
Comment (glennsong09) [reply release_docs/RELEASE_PROCESS.md]: Fixed.

File: release_docs/RELEASE_PROCESS.md (hunk 14 above)
Comment (glennsong09) [reply release_docs/RELEASE_PROCESS.md]: Updated.

File: release_docs/RELEASE_PROCESS.md (hunk 15 above)
Comment (glennsong09) [reply release_docs/RELEASE_PROCESS.md]: Updated.

File: release_docs/RELEASE_PROCESS.md (hunk 16 above)
Comment (glennsong09) [reply release_docs/RELEASE_PROCESS.md]: I tried changing how the formatting of that section looks. Please let me know if this is better.

File: release_docs/RELEASE_PROCESS.md (hunk 1 above)
Comment (glennsong09) [reply release_docs/RELEASE_PROCESS.md]: This has also been updated.

File: release_docs/RELEASE_PROCESS.md (hunk 13 above)
Comment (glennsong09) [reply release_docs/RELEASE_PROCESS.md]: Hi Binh-Minh, I tried updating this link, but it didn't seem to pass the links check once I did, so I have reverted the change.

File: release_docs/RELEASE_PROCESS.md [hunk 17]
```diff
@@ -0,0 +1,159 @@
...
+
+### 6. Prepare Release Branch (Release Manager)
+1. Get the release branch ready for pre-release testing and packaging.
+2. For all release preparation operations, the release coordinator will clone and push directly to canonical HDF5:
+    - `$ git clone ​https://github.com/HDFGroup/hdf5.git`
+3. Create a release preparation branch (hdf5_X_Y_Z) by branching off of the support branch (hdf5_X_Y):
+    - `$ git checkout hdf5_X_Y`
+    - `$ git checkout -b hdf5_X_Y_Z`
+    - or create the new branch in GitHub GUI.
+4. Check that required CMake files point to the specific versions of the third-party software (szip and zlib) that they depend on.
```
Comment (byrnHDF) [inline release_docs/RELEASE_PROCESS.md]: Change "(szip and zlib)" to "(szip, zlib and plugins)"

Comment (glennsong09) [reply release_docs/RELEASE_PROCESS.md]: Updated.

File: release_docs/RELEASE_PROCESS.md [hunk 18]
```diff
@@ -0,0 +1,163 @@
...
+
+Maintenance releases are always backward compatible with regards to the HDF5 file format, meaning that:
+- New libraries and command line utilities can access HDF5 files created by the previous versions of the libraries.
+
+Maintenance releases are always forward compatible with regards to the HDF5 file format, meaning that:
+- HDF5 libraries and command line utilities can access files created by future maintenance versions of the library.
+Note that maintenance releases are NOT guaranteed to be interface-compatible, meaning that, on occasion, application source code will need updated and re-compiled against a new maintenance release when the interface changes. Interface changes are only made when absolutely necessary as deemed by the HDF5 product manager(s), and interface compatibility reports are published with each release to inform customers and users of any incompatibilities in the interface.
+
+For more information on the HDF5 versioning and backward and forward compatibility issues, see the [API Compatibility Macros](https://docs.hdfgroup.org/hdf5/v1_14/api-compat-macros.html) on the public website.
```
Comment (lrknox) [inline release_docs/RELEASE_PROCESS.md]:
```suggestion
For more information on the HDF5 versioning and backward and forward compatibility issues, see the [API Compatibility Macros](https://docs.hdfgroup.org/hdf5/v1_14/v1_14_4/api-compat-macros.html) on the public website.
```

Comment (lrknox) [reply release_docs/RELEASE_PROCESS.md]: This will fix the broken link for now, but may or may not move in the future

**Review by lrknox — APPROVED** [review]

**Review by byrnHDF — APPROVED** [review]

**Pull request merged**